compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -4416,7 +4416,7 @@ private void checkAnnotConstantExpression(BLangExpression expression) {
                 ((BLangListConstructorExpr) expression).exprs.forEach(this::checkAnnotConstantExpression);
                 break;
             default:
-                dlog.error(expression.pos, DiagnosticErrorCode.EXPRESSION_IS_NOT_A_CONSTANT_EXPRESSION);
+                dlog.error(expression.pos, DiagnosticErrorCode.ANNOTATION_ELEMENT_SHOULD_BE_LITERAL);
```

Comment:
Please refer to https://github.com/ballerina-platform/ballerina-lang/issues/38746#issuecomment-1594437887